vanilla/model_views.py

```diff
@@ -133,7 +134,9 @@ def paginate_queryset(self, queryset, page_size):
             return paginator.page(page_number)
         except InvalidPage as exc:
             msg = 'Invalid page (%s): %s'
-            raise Http404(_(msg % (page_number, text(exc))))
+            if sys.version_info[0] == 2:
```

Comment:
Great. I’d suggest we push the branch out of the method, and into the module.

Eg. Start this module with..

```python
import sys

text = unicode if sys.version_info[0] == 2 else str
```